test/testconstructors.cpp

```diff
@@ -489,6 +491,50 @@ class TestConstructors : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void simple16() {
+        check("struct S {\n"
+              "    int i{};\n"
+              "    S() = default;\n"
+              "    S(const S & s) {}\n"
+              "    S& operator=(const S & s) { return *this; }\n"
+              "};\n");
+        ASSERT_EQUALS("[test.cpp:4]: (warning) Member variable 'S::i' is not initialized in the constructor.\n"
```

Comment:
as there is a c++11 initializer.. I feel we should not complain that it's not initialized in the copy constructor. At least there is not a uninitialized member variable.

hmm.. I think it seems noisy to warn about all members that are not copied in operator= and copy constructors. you probably want to copy 99% of the members but then there are members sometimes that you just want to assign default values for. I know it's not your code but I wonder what we were thinking..

Reply:
Looking at the existing test cases. we used to make the warning inconclusive if the member has a constructor:
https://github.com/danmar/cppcheck/pull/3748/files#diff-7b4200e4834dc91e02928bb713c711d378f93d4c6b42cc95102015234153fd81R1361
We probably should do the same for a simple member with an initializer.

Reply:
> Looking at the existing test cases. we used to make the warning inconclusive if the member has a constructor:

I don't see which existing test case you talk about

Reply:
If we simplify the example in 10360:

```
class Derived
{
public:
  Derived(const Derived& Src) {}
  std::vector<int> m_Arr;
};
```
Cppcheck complains:

10360.cpp:5:3: warning: inconclusive: Member variable 'Derived::m_Arr' is not assigned in the copy constructor. Should it be copied? [uninitMemberVar]

This makes no sense to me! It is totally a FP. The `uninitMemberVar` is about uninitialized variables. m_Arr is a std::vector and therefore it will not be uninitialized. I feel that `uninitMemberVar` should really not be used here. it seems it was borrowed for some other checking. It was probably me that did this and I deeply regret that :-(. If we just want to complain about some assignment issues I think that is a totally different thing and it should have a very different error id that does not say "uninit".

Reply:
The link above was supposed to point [here](https://github.com/danmar/cppcheck/blob/main/test/testconstructors.cpp#L1316), but your example above is equivalent. I agree about the unfortunate `uninitMemberVar` ID. At least the message text is different in the inconclusive case.